alicloud/resource_alicloud_ram_account_password_policy.go

```diff
@@ -0,0 +1,222 @@
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"minimum_password_length": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      default_minimum_password_length,
+				ValidateFunc: intBetween(8, 32),
+			},
+			"require_lowercase_characters": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  default_require_lowercase_characters,
+			},
+			"require_uppercase_characters": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  default_require_uppercase_characters,
+			},
+			"require_numbers": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  default_require_numbers,
+			},
+			"require_symbols": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  default_require_symbols,
+			},
+			"hard_expiry": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  default_hard_expiry,
+			},
+			"max_password_age": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      default_max_password_age,
+				ValidateFunc: intBetween(0, 1095),
+			},
+			"password_reuse_prevention": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      default_password_reuse_prevention,
+				ValidateFunc: intBetween(0, 24),
+			},
+			"max_login_attempts": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      default_max_login_attempts,
+				ValidateFunc: intBetween(0, 32),
+			},
+		},
+	}
+}
+
+func resourceAlicloudRamAccountPasswordPolicyUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	request := ram.CreateSetPasswordPolicyRequest()
+	if v, ok := d.GetOk("minimum_password_length"); ok {
+		request.MinimumPasswordLength = requests.NewInteger(v.(int))
+	}
+	// due to lack of GetOkExists() in old Terraform included in vendor dir
+	// GetOk() doesn't distinguish beetwen false/0 and empty, so in else clause set 0 or false
+	if v, ok := d.GetOk("require_lowercase_characters"); ok {
+		request.RequireLowercaseCharacters = requests.NewBoolean(v.(bool))
+	} else {
+		request.RequireLowercaseCharacters = requests.NewBoolean(false)
+	}
+	if v, ok := d.GetOk("require_uppercase_characters"); ok {
+		request.RequireUppercaseCharacters = requests.NewBoolean(v.(bool))
+	} else {
+		request.RequireUppercaseCharacters = requests.NewBoolean(false)
+	}
+	if v, ok := d.GetOk("require_numbers"); ok {
+		request.RequireNumbers = requests.NewBoolean(v.(bool))
+	} else {
+		request.RequireNumbers = requests.NewBoolean(false)
+	}
+	if v, ok := d.GetOk("require_symbols"); ok {
+		request.RequireSymbols = requests.NewBoolean(v.(bool))
+	} else {
+		request.RequireSymbols = requests.NewBoolean(false)
+	}
+	if v, ok := d.GetOk("max_login_attempts"); ok {
+		request.MaxLoginAttemps = requests.NewInteger(v.(int))
+	} else {
+		request.MaxLoginAttemps = requests.NewInteger(0)
```

Comment:
Value 5 is the default for `MaxLoginAttemps` (for 5 condition will be evaluated as true). In this piece of code, I provide a possibility for assigning 0 to this property (valid range is [0-32], where 0 means disable). I have to do it in this way due to the limitation of `GetOk()` (treats false, 0, etc. value the same as empty) with GetOkExists() it would be more elegant, but one's not available.